tests/test_filters.py

```diff
@@ -92,29 +94,141 @@ def test_last_turns():
     assert last_turns(colang_history, 2) == colang_history
 
 
+def _build_test_string(parts: List[Union[str, Tuple[str, int]]]) -> str:
+    """Builds a test string from a list of parts.
+
+    Each part can be a literal string or a (character, count) tuple.
+    Example: [("a", 3), "[START]", ("b", 5)] -> "aaa[START]bbbbb"
+    """
+    result = []
+    for part in parts:
+        if isinstance(part, str):
+            result.append(part)
+        elif isinstance(part, tuple) and len(part) == 2:
+            char, count = part
+            result.append(char * count)
+        else:
+            raise TypeError(f"Invalid part type in _build_test_string: {part}")
+    return "".join(result)
+
+
 @pytest.mark.parametrize(
```

Comment:
Also add out-of-order cases like "[END] [START] [START]"

Reply:
Totally, just added 👍🏻